### ISSUE TITLE
Analyzer Misalignment: [SupportedOSPlatform] on Process.Start incorrectly restricts use to maccatalyst

### ISSUE THREAD

**Issue by evoronkin** [dotnet/roslyn-analyzers#7497]
### Analyzer

**Diagnostic ID**: [CA1416](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416): `Validate platform compatibility`

### Analyzer source

Environment Details

.NET SDK Version: 8.0.x

Target Framework: net8.0

IDE/Tooling: Visual Studio 2022 (or CLI with dotnet commands)

Platform: Windows/Linux/macOS (as applicable)

### Describe the bug
Problem became visible from the introduction of this change
**Support Process.Start() on MacCatalyst** https://github.com/dotnet/runtime/pull/61507
When using Process.Start in a project targeting net8.0, the platform compatibility analyzer (CA1416) incorrectly emits a warning that the method is only supported on maccatalyst, even though it works on Windows, Linux, and macOS. This behavior seems to misinterpret the [SupportedOSPlatform("maccatalyst")] attribute as a restriction rather than an indication of additional support.

### Steps To Reproduce

1. Create a new .NET 8 project targeting net8.0.

2. Add code to call Process.Start, e.g.:

if (OperatingSystem.IsWindows() || OperatingSystem.IsLinux() || OperatingSystem.IsMacOS())
{
    Process.Start("your_program");
}

3. Build the project and observe the CA1416 warning.

Reproducible Code:

using System.Diagnostics;

class Program
{
    static void Main()
    {
        if (OperatingSystem.IsWindows() || OperatingSystem.IsLinux() || OperatingSystem.IsMacOS())
        {
            Process.Start("notepad");
        }
    }
}

.csproj Configuration:

<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <RuntimeIdentifiers>win-x64;linux-x64;osx-x64</RuntimeIdentifiers>
  </PropertyGroup>
</Project>

### Expected behavior
The analyzer should not emit warnings for Process.Start when:
1. The runtime platform is explicitly checked (e.g., OperatingSystem.IsWindows()).
2. A method guarded with [SupportedOSPlatformGuard] is used.
3. The target framework is net8.0 with runtime identifiers for win, linux, and osx.

### Actual behavior
The analyzer emits the following warning:

CA1416: This call site is reachable on all platforms. 'Process.Start(string)' is only supported on: 'maccatalyst'.

Even after:

1. Adding explicit runtime checks (e.g., if (OperatingSystem.IsWindows()) { Process.Start(...) }).
2. Using a SupportedOSPlatformGuard method.
3. Ensuring correct runtime identifiers (win, linux, osx) in the project configuration.

This prevents Process.Start from being used without suppressing the warning manually.
## Additional context

Additional Context

This misalignment forces developers to either suppress warnings manually or avoid using Process.Start altogether, despite its functional support across platforms. The expected behavior is for the analyzer to respect runtime checks or platform guards.

Similar Issues or Documentation:

Link to the documentation on [SupportedOSPlatform]: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.versioning.supportedosplatformattribute

Link to the documentation on [SupportedOSPlatformGuard]: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.versioning.supportedosplatformguardattribute

**Comment by buyaa-n** [comment]
I see `[SupportedOSPlatformAttribute("maccatalyst")]` attribute added to `Start()` and to many other APIs with https://github.com/dotnet/runtime/pull/61507 which is wrong:
```diff
        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("ios")]
        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("tvos")]
+       [System.Runtime.Versioning.SupportedOSPlatformAttribute("maccatalyst")]
        public bool Start() { throw null; }
```
Previous annotation expressed that `Process.Start()` was unsupported on `ios` and `tvos` and supported on all other, now after adding `[SupportedOSPlatformAttribute("maccatalyst")]` it became only supported on `maccatalyst` and unsupported on all other. @simonrozsival please remove the `[SupportedOSPlatformAttribute("maccatalyst")]` from all APIs that had: 
```cs
        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("ios")]
        [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("tvos")]
```
this means the API is unsupported on `ios` and `tvos` and supported on all other platforms including `maccatalyst`
